src/targets/gpu/convolution.cpp

```diff
@@ -148,35 +228,54 @@ shape miopen_convolution::find(context& ctx, const shape& output_shape, std::vec
     solution_id = solutions.front().solution_id;
 
     return shape{shape::int8_type, {perf.memory}};
+#endif
 }
 
 void miopen_convolution::finalize(context& ctx,
                                   const shape& output_shape,
-                                  std::vector<shape> inputs)
+                                  const std::vector<shape>& inputs)
 {
-    if(cd == nullptr)
-        cd = make_conv(op);
-    if(solution_id == 0)
+#ifdef MIGRAPHX_HAS_FIND_2_API
     {
-        // Check that workspace hasn't changed
-        auto size = inputs.at(2).bytes();
-        auto ws   = find(ctx, output_shape, inputs);
-        if(ws.bytes() > size)
-            MIGRAPHX_THROW("MIOpen Convolution: workspace has changed during finalization.");
+        (void)(ctx); // avoid warnings
+        (void)(output_shape);
+        (void)(inputs);
+        // load solution
```

Comment:
```suggestion
        if (solution_ptr != nullptr)
            return;
        // load solution
```

Reply:
> If your typical use case is prepare -> finalize in a new process -> not save second time, I would suggest releasing ptr after saving

So the process during compile and serialize is : `find()-->finalize()-->serialize()`.  During `find` it would create binary member variable for `miopen_convolution` class to save the solution as binary.  This would remain in memory until serialize is called. At that time, the contents of `miopen_convolution` gets written to some file. 

it will call `miopenDestroySolution` automatically if the `miopen_convolution` object goes out of scope.  If it is in the same process then, it wouldn't destroy and ptr would remain valid and during `finalize` it would avoid `Loading`. 

I am not sure if it is possible to clear/release binary solution_object member variable once it has been loaded in memory during finalize to avoid having duplicates.

Reply:
Why won't you just create binary solution in serialize? And on deserialization you just load it. This way you don't need a member field for binary at all, only storing it as a temporary.

Reply:
We would need to create a custom serialized class to do this. I think what Umang is doing now is fine and we could address this in a future PR.